File: frontend/src/app/models/device/device-type.ts
export enum DeviceType {
  LAN_CLIENT = 'LAN_CLIENT',
  WIFI_CLIENT = 'WIFI_CLIENT',
  NETWORK_DEVICE = 'NETWORK_DEVICE',
}

export const DEVICE_TYPE_LABELS: Record<DeviceType, string> = {
  [DeviceType.LAN_CLIENT]: 'LAN Client',
  [DeviceType.WIFI_CLIENT]: 'WiFi Client',
  [DeviceType.NETWORK_DEVICE]: 'Network Infrastructure',
};

File: web-api/src/main/java/pl/sgorski/nethelt/webapi/features/device/domain/DeviceType.java
package pl.sgorski.nethelt.webapi.features.device.domain;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
public enum DeviceType {
  LAN_CLIENT("Client LAN"),
  WIFI_CLIENT("Client WiFi"),
  NETWORK_DEVICE("Network Infrastructure");

  private final String displayName;
}
